flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -358,15 +360,15 @@ public static TaskExecutor startTaskManager(
 				remoteAddress,
 				localCommunicationOnly);
 
-		TaskManagerMetricGroup taskManagerMetricGroup = MetricUtils.instantiateTaskManagerMetricGroup(
+		Tuple2<TaskManagerMetricGroup, MetricGroup> taskManagerMetricGroup = MetricUtils.instantiateTaskManagerMetricGroup(
```

Comment:
I would consider moving:
```
MetricGroup statusGroup = taskManagerMetricGroup.addGroup(METRIC_GROUP_STATUS_NAME);
// Initialize the TM metrics			// Initialize the TM metrics
instantiateStatusMetrics(statusGroup);			instantiateStatusMetrics(statusGroup);
```
out of `MetricUtils.instantiateTaskManagerMetricGroup` into this method here
instead of having `Tuple2` return type, imho, it will be more readable.

Reply:
This may be something we have to live with. The `MetricUtils` exists so that JM/TM are roughly in sync with each other in what metrics they expose and how the scope works.

Moving the status group initialization of the TM out of here makes this a lot less obvious.